scripts/checks/ruleof6/tests/test_false_positives.py

```
@@ -0,0 +1,243 @@
+        for name in false_positive_names:
+            assert name not in function_names, f"'{name}' should NOT be detected as a function declaration"
+
+        # Should detect real function declarations
+        expected_functions = ['testFunctionCalls', 'realFunction', 'arrowFunction']
+        for name in expected_functions:
+            assert name in function_names, f"'{name}' should be detected as a function declaration"
+
+    def test_object_arguments_counted_correctly(self):
+        """Test that object arguments are counted as single arguments, not multiple."""
+        # Test single object argument
+        single_obj_code = """
```

Ruff (0.13.2)

Local variable `single_obj_code` is assigned to but never used

Remove assignment to unused variable `single_obj_code`

(F841)